src/utilities/parameters.ts

```diff
@@ -1,26 +1,43 @@
 import { omitBy } from 'lodash-es';
 import { isEmpty } from './generic';
 
-export function getArgument(value: Argument, schema: ValueSchema, defaultValue?: Argument): any {
+// DOT:
+// Value comes from user if arg is not null or undefined (orange dot)
+// Otherwise, value comes from mission if default arg is not undefined (green dot)
+// Otherwise, there is no value and we don't need to show a dot (no dot)
+
+/**
+ * Derive argument given input value, value schema, and optional default value.
+ * Returns the derived value and the source of the value which follows this logic:
+ * if the value is not null or undefined: "user"
+ * else if the default value is not undefined: "mission"
+ * otherwise there is no value so there is no value source: "none"
+ * @param value
```

Comment:
Can we remove the `@param` and `@returns`? They are already with the TypeScript types.